sdk/formrecognizer/azure-ai-formrecognizer/tests/test_dmac_compose_model.py

```diff
@@ -114,11 +114,12 @@ def test_poller_metadata(self, client, formrecognizer_storage_container_sas_url,
         model_2 = poller.result()
 
         poller = client.begin_compose_model([model_1.model_id, model_2.model_id])
-        assert poller.operation_id
-        assert poller.percent_completed is not None
         poller.result()
-        assert poller.operation_kind == "documentModelCompose"
-        assert poller.percent_completed == 100
-        assert poller.resource_location_url
-        assert poller.created_on
-        assert poller.last_updated_on
+        details = poller.details
```

Comment:
consider adding an isinstance check for the poller vs. protocol, it will at least alert you if our poller type stops conforming to the protocol.